packages/providers/src/provider.ts

```diff
@@ -279,17 +282,23 @@ export default class Provider {
     timeout: undefined,
     cacheUtxo: undefined,
     fetch: undefined,
+    retryOptions: undefined,
   };
 
   private static getFetchFn(options: ProviderOptions) {
-    return options.fetch !== undefined
-      ? options.fetch
-      : (url: string, request: FetchRequestOptions) =>
-          fetch(url, {
-            ...request,
-            signal:
-              options.timeout !== undefined ? AbortSignal.timeout(options.timeout) : undefined,
-          });
+    const fetchFn =
+      options.fetch !== undefined
+        ? options.fetch
+        : (url: string, request: FetchRequestOptions) =>
+            fetch(url, {
+              ...request,
+              signal:
+                options.timeout !== undefined ? AbortSignal.timeout(options.timeout) : undefined,
+            });
+
+    type FetchParams = Parameters<NonNullable<ProviderOptions['fetch']>>;
+
+    return (...args: FetchParams) => retrier(() => fetchFn(...args), options.retryOptions);
```

Comment:
These points make this code chunk unnecessarily unreadable.
 1. Lots of lambdas
 1. Ternary conditions
 1. Anonymous returns

Reply:
True that. I refactored it a bit. How does it look now?

Reply:
Much better, thanks.

I've left an extra NIT/suggestion [here](https://github.com/FuelLabs/fuels-ts/pull/1474#discussion_r1435262592) for your appreciation.